### ISSUE TITLE
Issue with limiting user field by Role, deleting that role, and not being able to unselect the role from being limited

### ISSUE THREAD

**Issue by ckpicker** [pods-framework/pods#1902]
I've created a Relationship field named 'user_account' in my 'members' Pod that is related to all WP Users. (http://d.pr/i/HNHj)

I have a Pods UI Plugin that I've created and am using for this site (code: https://gist.github.com/ckpicker/24d11338bcb2ee5738a4 - see 'members_page' function).

The issue is that no users are displaying when I view the dropdown from the Pods UI plugin (http://d.pr/i/BiUX), but I currently have 4 users in my WP instance.

Any ideas on this one?

**Comment by jimtrue** [comment]
Do you have the user field set to show only users of a specific capability?

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
No, here's what the 'Additional Field Options' tab looks like for that field: http://d.pr/i/Jyoa

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
However, this seems to work in my Pods UI plugin if I use the older 'pods_ui_manage' method instead of 'pods_ui.'

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
Hmm, I'm not seeing anything there in the options or the code that would cause that. Though you don't need edit_fields/add_fields in the options array, that's handled by `'fields' => array( 'add' => $add_fields, 'edit' => $edit_fields )`

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
Good catch!  I deleted those extra options, but that didn't fix the issue.  I also tried using an autocomplete field with no luck either.

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
Any ideas on this one?  I still can't figure out what's going on.

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
Yeah, this is on my list for this week, thanks for being patient.

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
Can I get access the project itself to do some testing?

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
No problem!  Just sent you a hangout with the info.  Thanks!

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
There is indeed an issue here. What happened was you created the user_account field and selected the 'members' role to be limited to for which users would show up. Then at some point, that role was deleted, and a new one of a different name similar to 'members' was added, but they didn't have the same name. What this caused was the inability to unselect the role from the field editor. Even though it appeared that no roles were limited, there was a 'members' value stored in the options array for that field which could not be toggled off.

We'll get this fixed during the field editor revamp of 3.0 for loop fields.

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by jimtrue** [comment]
If I remember correctly, I renamed the role.  Thanks for tracking this one down!

[--- Linked with ASANA.com using Unito.io
#commenter Jim True
---[aa]

**Comment by sc0ttkclark** [comment]
Let's get a retest on this to check if the DFV revamp resolved this issue in Pods 2.8.

**Comment by sc0ttkclark** [comment]
Ready for QA test

**Comment by pdclark** [comment]
QA mostly didn't pass:
- Deleting the role caused the toggle to no longer appear in the relationship field options.
- Choosing a new role and saving didn't remove the old deleted role from the array.

<img width="597" alt="Screen Shot 2021-09-16 at 3 31 37 PM" src="https://user-images.githubusercontent.com/126889/133694099-6540ef14-7b7a-4331-838d-bf3f51ae6869.png">
<img width="514" alt="Screen Shot 2021-09-16 at 3 31 23 PM" src="https://user-images.githubusercontent.com/126889/133694108-40c71841-0b70-421a-a9c5-579d98fb3204.png">
<img width="277" alt="Screen Shot 2021-09-16 at 3 33 26 PM" src="https://user-images.githubusercontent.com/126889/133694140-a059da1d-7976-4bfa-8ad1-a90cc6cdee95.png">

**Comment by sc0ttkclark** [comment]
Something for the backlog but this would be good to sort out from the DFV side of the field.
